src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -142,13 +139,30 @@ internal async void ShutdownWebhostLanguageWorkerChannels()
             await _webHostLanguageWorkerChannelManager?.ShutdownChannelsAsync();
         }
 
-        private void StartWorkerProcesses(int startIndex, Action startAction)
+        private void StartWorkerProcesses(int startIndex, Func<Task> startAction)
         {
-            for (var count = startIndex; count < _maxProcessCount; count++)
+            Task.Run(async () =>
             {
-                startAction = startAction.Debounce(_processStartCancellationToken.Token, count * _debounceMilliSeconds);
-                startAction();
-            }
+                for (var count = startIndex; count < _maxProcessCount
+                    && !_processStartCancellationToken.IsCancellationRequested; count++)
+                {
+                    try
+                    {
+                        await startAction();
+                        await Task.Delay(_processStartupInterval);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, $"Failed to start a new language worker for runtime: {_workerRuntime}.");
+                    }
+                }
+
+                if (State != FunctionInvocationDispatcherState.Initialized)
```

Comment:
Checking initialized state is handled here: https://github.com/Azure/azure-functions-host/blob/1927f5fd736931d2a6407b84d35927f2404eb8dc/src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs#L109-L123 

Safe to remove the check here.

Reply:
The check is removed,